libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMp4aPayloadReader.java

```diff
@@ -0,0 +1,157 @@
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Parses an MP4A-LATM byte stream carried on RTP packets, and extracts MP4A-LATM Access Units.
+ * Refer to RFC3016 for more details.
+ */
+@UnstableApi
+/* package */ final class RtpMp4aPayloadReader  implements RtpPayloadReader {
+  private static final String TAG = "RtpMp4aLatmReader";
+
+  private static final String PARAMETER_MP4A_CONFIG = "config";
+
+  private final RtpPayloadFormat payloadFormat;
+  private @MonotonicNonNull TrackOutput trackOutput;
+  private long firstReceivedTimestamp;
+  /** The combined size of a sample that is fragmented into multiple subFrames. */
+  private int fragmentedSampleSizeBytes;
+  private long startTimeOffsetUs;
+
+  /** Creates an instance. */
+  public RtpMp4aPayloadReader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    fragmentedSampleSizeBytes = 0;
+    // The start time offset must be 0 until the first seek.
+    startTimeOffsetUs = 0;
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    checkState(firstReceivedTimestamp == C.TIME_UNSET);
+    firstReceivedTimestamp = timestamp;
+  }
+
+  @Override
+  public void consume(
+      ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    checkStateNotNull(trackOutput);
+
+    int sampleOffset = 0;
+    int numSubFrames = getNumOfSubframesFromMpeg4AudioConfig(payloadFormat.fmtpParameters);
+    for (int subFrame = 0; subFrame <= numSubFrames; subFrame++) {
+      int sampleLength = 0;
+
+      /* Each subframe starts with a variable length encoding */
+      for (; sampleOffset < data.bytesLeft(); sampleOffset++) {
+        sampleLength += data.getData()[sampleOffset] & 0xff;
+        if ((data.getData()[sampleOffset] & 0xff) != 0xff) {
+          break;
+        }
+      }
+      sampleOffset++;
+      data.setPosition(sampleOffset);
+
+      // Write the audio sample
+      trackOutput.sampleData(data, sampleLength);
+      sampleOffset += sampleLength;
+      fragmentedSampleSizeBytes += sampleLength;
+    }
+    if (rtpMarker) {
+      long timeUs =
+          toSampleUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp, payloadFormat.clockRate);
+      trackOutput.sampleMetadata(
+          timeUs, C.BUFFER_FLAG_KEY_FRAME, fragmentedSampleSizeBytes, 0, null);
+      fragmentedSampleSizeBytes = 0;
+    }
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
+    firstReceivedTimestamp = nextRtpTimestamp;
+    fragmentedSampleSizeBytes = 0;
+    startTimeOffsetUs = timeUs;
+  }
+
+  // Internal methods.
+  private static long toSampleUs(
```

Comment:
remove and use `import static androidx.media3.exoplayer.rtsp.reader.RtpReaderUtils.toSampleTimeUs;`